syntax = "proto3";

import "qroma-lights-types.proto";


message SaveQromaStrip_CurrentState {
  QromaStrip_WS2812FX_StripIndex stripIndex = 1;
}


message QromaStripCommand {
  QromaStrip_WS2812FX_StripIndex stripIndex = 1;
  oneof command {
    uint32 setQromaStripBrightness = 10; // 0-255
    QromaStrip_WS2812FX_Animation setQromaStripAnimation = 11;
    QromaStrip_WS2812FX_IoSettings setQromaStripIoSettings = 12;
  }
}


message SetQromaDeviceName {
  string name = 1;
}


// not sure how I feel about this, but let's try it out, at least
enum NoArgCommands_QromaLightsDeviceCommand {
  NacQlc_NotSet = 0;
  NacQlc_UseCurrentSetupOnStartup = 1;
  NacQlc_RestartQromaDevice = 2;
  NacQlc_FactoryResetQromaDevice = 3;
  NacQlc_RequestQromaLightsConfig = 4;
}


message QromaLightsDeviceCommand {
  oneof command {
    NoArgCommands_QromaLightsDeviceCommand noArgCommand = 1;
    SetQromaDeviceName setQromaDeviceName = 2;
    QromaStripCommand qromaStripCommand = 3;
  }
}


message QromaLightsDeviceError {
  string message = 1;
}


message QromaLightsResponse {
  oneof response {
    QromaLightsDeviceConfig qromaLightsConfigResponse = 1;
    QromaLightsDeviceError error = 2;
  }
}
